tests/drivers/rtc/rtc_api_helpers/testcase.yaml

```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2022 Bjarki Arge Andreasen
+# SPDX-License-Identifier: Apache-2.0
+
+tests:
+  drivers.rtc.rtc_api:
+    tags: drivers rtc api helpers
+    platform_allow: native_posix
```

Comment:
This test is probably going to need to also run on at least variations of all architectures, e.g. mips, arm, etc. and if they have different endian options.